virtualizarr/kerchunk.py

```diff
@@ -68,25 +67,15 @@ def _automatically_determine_filetype(filepath: str) -> str:
     file_extension = Path(filepath).suffix
 
     if file_extension == ".nc":
-        # checks if netCDF library is installed.
-        # It currently is not a requirement in the pyproj.toml.
-
-        if importlib.util.find_spec("netCDF4") is None:
-            raise ImportError(
-                "netCDF4 library is required to determine NetCDF file type."
-            )
-
-        import netCDF4
-
-        with netCDF4.Dataset(filepath, "r") as dataset:
-            if dataset.data_model == "NETCDF4":
-                filetype = "netCDF4"
-            elif dataset.data_model == "NETCDF3_CLASSIC":
-                filetype = "netCDF3"
```

Comment:
This function should return an [enum](https://docs.python.org/3/library/enum.html) not a string.

Everything in this library should be super strictly typed.

Reply:
Should these be similar to how it's done in `pangeo-forge-recipes`, ie.
```
class FileType(AutoName):
    netcdf3 = auto()
    netcdf4 = auto()
    grib = auto()


...

filetype = FileType("netcdf3")
```